src/python/pants/backend/experimental/python/lockfile.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import logging
+from typing import cast
+
+from pants.backend.python.subsystems.python_tool_base import PythonToolBase
+from pants.backend.python.target_types import ConsoleScript, PythonRequirementsField
+from pants.backend.python.util_rules.interpreter_constraints import InterpreterConstraints
+from pants.backend.python.util_rules.pex import PexRequest, PexRequirements, VenvPex, VenvPexProcess
+from pants.engine.fs import CreateDigest, Digest, FileContent, Workspace
+from pants.engine.goal import Goal, GoalSubsystem
+from pants.engine.process import ProcessResult
+from pants.engine.rules import Get, MultiGet, collect_rules, goal_rule
+from pants.engine.target import Targets
+from pants.util.strutil import pluralize
+
+logger = logging.getLogger(__name__)
+
+
+class PipCompile(PythonToolBase):
```

Comment:
The tool is pip-tools so this should be called PipTools, and ditto the scope. 

The fact that we only use pip-compile is orthogonal to defining how we download pip-tools...